eng/common/scripts/Update-DocsMsMetadata.ps1

```diff
@@ -0,0 +1,150 @@
+in build artifacts under packages/PackageInfo/<package-name>.json. Can also be
+a single item.
+
+.PARAMETER DocRepoLocation 
+Location of the root of the docs.microsoft.com reference doc location. Further
+path information is provided by $GetDocsMsMetadataForPackageFn
+
+.PARAMETER Language
+Programming language to supply to metadata
+
+.PARAMETER RepoId
+GitHub repository ID of the SDK. Typically of the form: 'Azure/azure-sdk-for-js'
+
+#>
+
+param(
+  [Parameter(Mandatory = $true)]
+  [array]$PackageInfoJsonLocations,
+  
+  [Parameter(Mandatory = $true)]
+  [string]$DocRepoLocation, 
+
+  [Parameter(Mandatory = $true)]
+  [string]$Language,
+
+  [Parameter(Mandatory = $true)]
+  [string]$RepoId
+)
+
+. (Join-Path $PSScriptRoot common.ps1)
+
+$releaseReplaceRegex = "(https://github.com/$RepoId/(?:blob|tree)/)(?:master|main)"
+$TITLE_REGEX = "(\#\s+(?<filetitle>Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C)))"
+
+function GetAdjustedReadmeContent($ReadmeContent, $PackageInfo, $PackageMetadata) {
+  # Normalize service name "Key Vault" -> "keyvault"
+  # TODO: Use taxonomy for service name -- https://github.com/Azure/azure-sdk-tools/issues/1442
+  # probably from metadata
+  $service = $PackageMetadata.ServiceName.ToLower().Replace(" ", "")
+  # Generate the release tag for use in link substitution
+  $tag = "$($PackageInfo.Name)_$($PackageInfo.Version)"
+  $date = Get-Date -Format "MM/dd/yyyy"
+
+
+  $foundTitle = ""
+  if ($ReadmeContent -match $TITLE_REGEX) {
+    $ReadmeContent = $ReadmeContent -replace $TITLE_REGEX, "`${0} - Version $($PackageInfo.Version) `n"
+    $foundTitle = $matches["filetitle"]
+  }
+
+  $replacementPattern = "`${1}$tag"
+  $ReadmeContent = $ReadmeContent -replace $releaseReplaceRegex, $replacementPattern
+  
+  $header = @"
+---
+title: $foundTitle
+keywords: Azure, $Language, SDK, API, $($PackageInfo.Name), $service
+author: maggiepint
+ms.author: magpint
+ms.date: $date
+ms.topic: article
+ms.prod: azure
+ms.technology: azure
+ms.devlang: $Language
+ms.service: $service
+---
+
+"@
+
+  return "$header`n$ReadmeContent"
+}
+
+function UpdateDocsMsMetadataForPackage($packageInfoJsonLocation) { 
+  $packageInfoJson = Get-Content $packageInfoJsonLocation -Raw
+  $packageInfo = ConvertFrom-Json $packageInfoJson
+
+  $originalVersion = [AzureEngSemanticVersion]::ParseVersionString($packageInfo.Version)
+  if ($packageInfo.DevVersion) {
+    # If the package is of a dev version there may be language-specific needs to 
+    # specify the appropriate version. For example, in the case of JS, the dev 
+    # version is always 'dev' when interacting with NPM.
+    if ($GetDocsMsLanguageSpecificPackageInfo -and (Test-Path "Function:$GetDocsMsLanguageSpecificPackageInfo")) { 
+      $packageInfo = &$GetDocsMsLanguageSpecificPackageInfo $packageInfo
+    }
+  }
+
+  $packageMetadataArray = (Get-CSVMetadata).Where({ $_.Package -eq $packageInfo.Name })
```

Comment:
We probably need to filter out "Hide=true" or you will get duplicate packages, in JS and Python in some cases. We also should think if we want to filter down to only "New=true" which I suspect we do because we mostly care about Track 2.